pkg/kubectl-argo-rollouts/viewcontroller/viewcontroller.go

```diff
@@ -182,26 +185,55 @@ func (c *RolloutViewController) GetRolloutInfo() (*rollout.RolloutInfo, error) {
 		return nil, err
 	}
 
-	allReplicaSets, err := c.replicaSetLister.List(labels.Everything())
+	selector, err := metav1.LabelSelectorAsSelector(ro.Spec.Selector)
 	if err != nil {
 		return nil, err
 	}
 
-	allPods, err := c.podLister.List(labels.Everything())
+	// Get rs of rollouts
+	allReplicaSets, err := c.replicaSetLister.List(selector)
 	if err != nil {
 		return nil, err
 	}
 
-	allExps, err := c.experimentLister.List(labels.Everything())
+	// Get experiments and their rs
+	selector = labels.SelectorFromSet(map[string]string{
+		v1alpha1.DefaultRolloutLabelKey:          ro.Name,
+		v1alpha1.DefaultRolloutNamespaceLabelKey: ro.Namespace,
```

Comment:
I don't think we need `v1alpha1.DefaultRolloutNamespaceLabelKey` at all. The c.experimentLister c.rolloutLister, c.replicaSetLister are all already namespaced lister and so it wouldn't return anything outside of the current namespace.